File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        char[] splitters = {' ', '_', '-'};
        var phraseWords = phrase.Split(splitters, StringSplitOptions.RemoveEmptyEntries);

        return phraseWords
            .Aggregate(string.Empty, (acronym, word) => acronym + char.ToUpperInvariant(word[0]));
    }
}

File: python/acronym/acronym.py
import re
from functools import reduce


def abbreviate(words):
    words = re.sub('[-_]', ' ', words)

    acronym = reduce(lambda acc, word: acc + word[0].upper(), words.split(), '')
    # acronym = ''.join([word[0].upper() for word in words.split()])

    return acronym
